lib/pushy_client/protocol_handler.rb

```diff
@@ -411,11 +404,10 @@ def check_server_address(name, server_address)
     # MAX_BODY_SIZE. If they do, log a warning and return the params without the
     # stdout and stderr keys. If they do not, return the params hash unmodified.
     def validate_params(params = {})
-      stdout_bytes = params[:stdout].to_s.bytesize
-      stderr_bytes = params[:stderr].to_s.bytesize
-
-      if (stdout_bytes + stderr_bytes) > MAX_BODY_SIZE
-        Chef::Log.warn("Command output too long. Will not be sent to server.")
+      if (params.to_s.bytesize) > client.max_body_size.to_i
+        Chef::Log.warn("****************************************************")
+        Chef::Log.warn("Client packet size #{params.to_s.bytesize} is larger than the Client MAX_BODY_SIZE of #{client.max_body_size.to_i}")
+        Chef::Log.warn("****************************************************")
         params.delete_if { |k| [:stdout, :stderr].include? k }
```

Comment:
Wherever this code lands, we should not simply delete the message, but provide some indication in the packet that it had to be deleted. Otherwise it will be surprising to the user when nothing comes through. Even better, would be some partial result. Perhaps take the last N characters of the stdout and stderr, and prepend it with 'Message truncated because it exceeded size limit {max_body_size}'.  N probably should be some function of maxsize; perhaps (maxsize-256)/8 to allow for unicode and standard headers; there are a lot of possible heuristics.